src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/GerritMessageProvider.java

```diff
@@ -71,10 +115,29 @@ public String getBuildCompletedMessage(AbstractBuild build) {
      * @param build the build to complain about
      * @return the file comments, default is an empty list.
      */
-    public Collection<CommentedFile> getFileComments(AbstractBuild build) {
+    public Collection<CommentedFile> getFileComments(Run build) {
+        // If the overriding class still overrides the deprecated method, call that.
+        if (build instanceof AbstractBuild && thisOverrides("getFileComments", AbstractBuild.class)) {
+            return getFileComments((AbstractBuild)build);
+        }
         return Collections.emptyList();
     }
 
+    /**
+     * Provide any file comments.
+     *
+     * @param build the build to complain about
+     * @return the file comments, default is an empty list.
+     * @deprecated Use {@link #getFileComments(hudson.model.Run)}
+     */
+    @Deprecated
+    public Collection<CommentedFile> getFileComments(AbstractBuild build) {
+        if (thisOverrides("getFileComments", Run.class)) {
+            return getFileComments((Run)build);
+        }
+        return null;
```

Comment:
IIRC the previous contract was to return an empty list and not null?, see method above